### PR TITLE
feat(hooks): ready-to-fill falsified scaffold

### PR DESCRIPTION
**Pull request by devseunggwan**
## 요약

`output-block-falsify-advisory` 훅의 T1 deny / T2 ask 메시지에 실제 트리거된 옵션 라벨로 시드된 복붙 가능한 `Falsified: <label> — probe: ...` scaffold 줄을 추가했습니다. 매 재시도마다 형식을 처음부터 재구성해야 하는 비용을 없앱니다. 또한 이 훅은 stdout JSON으로 deny/ask를 신호하기 때문에 exit code만 보는 coarse fire-ledger 레코더에는 보이지 않아, block-event 텔레메트리(RICH fire-ledger 레코드, `decision=block/ask`)도 함께 기록하도록 추가했습니다.

Closes #787

## ⚠️ 헤드라인 — diminishing-returns 신호 (3배 초과, 반드시 먼저 읽어주세요)

이 PR은 **12라운드**의 Codex 리뷰를 거쳤습니다. praxis의 `codex-review-wrap` 스킬은 동일 `{file}:{region}`이 기본 임계값 `PRAXIS_DIMINISHING_RETURNS_N=4`를 넘으면 advisory를 발동하는데, `impl.py`의 `_has_falsified_line`(및 라벨 매칭 로직)은 그 **3배 이상**을 초과했습니다. 특히 **라운드 9~12는 4회 연속으로 새로운 우회(bypass) 사례**를 찾아냈고, **라운드 13은 Codex 리뷰가 실행 중 강제 종료(killed)되어 결과를 얻지 못한 채 중단**했습니다 — 즉 "새 발견 없음(convergence)"으로 종료된 라운드가 아직 한 번도 없습니다.

각 라운드의 수정이 이전 수정이 만든 새롭고 점점 더 좁아지는 엣지 케이스를 드러내는 패턴이 반복되었습니다(특히 라운드 10→12는 "라벨 뒤 경계 조건"이라는 같은 축에서 3회 연속 좁혀짐). 이는 string-matching 휴리스틱 접근 자체의 견고성에 한계가 있을 가능성을 시사합니다 — 리뷰어와 병합자는 이 PR을 병합용으로 승인하기 전에, 남은 잔여 우회 벡터가 있을 수 있다는 점을 감안해 주시기 바랍니다. (커밋 트레일러에도 `Not-tested:`로 동일 내용을 명시했습니다.)

## 전체 라운드별 발견 사항 (모두 라이브 프로브로 재현 확인 → 수정 → 회귀 테스트로 검증)

| 라운드 | 발견 | 수정 |
|---|---|---|
| 2~8 | placeholder 미기입 감지, 옵션별 line 삭제 우회, 라벨 텍스트에 placeholder 유사 문자열 포함 시 오탐, marker-less legacy 줄 오탐, 중복 줄 붙여넣기 우회, whitespace-only 중복 우회, 라벨에 delimiter 문자열 포함 시 permablock | placeholder scope를 `— probe:` 마커 이후로 한정, `required_count` 기반 카운트 검증, marker-less 줄 placeholder 스캔 제외, 텍스트 정규화 후 dedup, `rfind()`로 실제 delimiter 앵커링 |
| 9 (P1) | 서로 다른 문구의 두 줄이 모두 같은 옵션(A)만 가리켜도 count-only 검증이 silent-pass — 옵션 B는 전혀 검증 안 됨 | count 기반 → 각 clean line을 트리거 라벨과 매칭시켜 **라벨별 커버리지**로 전환 |
| 9 (P2) | evidence 텍스트 자체에 `— probe:`가 재등장하면 `rfind()`가 진짜 delimiter를 놓치고 그 앞의 미기입 placeholder를 스캔 범위 밖으로 밀어냄 | 매칭된 라벨 끝 지점부터 `find()`로 delimiter 탐색(라운드 8의 label-embedded-marker 케이스도 동시에 유지) |
| 10 (P2) | `"Run"` 라벨이 `"Runtime behavior..."`처럼 완전히 무관한 텍스트의 접두사로 우연히 매칭 | 라벨 뒤 비영숫자(non-alnum) 경계 요구 |
| 11 (P2) | 옵션 라벨 자체에 개행(`\n`)이 있으면 생성된 scaffold가 물리적으로 2줄로 쪼개져, 미기입 placeholder가 있는 두 번째 줄이 `Falsified:`로 시작하지 않아 스캔에서 완전히 빠짐 | 라벨을 트리거 파이프라인 진입 시점(`_t1_matching_labels`/`_t2_matching_labels`)에 whitespace 정규화 |
| 12 (P2) | `"Run"` vs `"Run now"`처럼 공백으로 구분된 서로 다른(무관한) 옵션도 비영숫자 경계를 통과해 라운드 10 수정을 우회 | 경계 조건을 "정확히 scaffold delimiter(`" — probe: "`) 또는 줄 끝"으로 강화 |
| 13 | (미완료 — Codex 리뷰가 실행 중 강제 종료됨, 결과 없음) | — |

Pre-commit verified: `bash tests/hooks/advisory-nudge/test_output_block_falsify_advisory.sh` → 88/88 passed (rebase 이후 재확인); `bash scripts/run-tests.sh` → ALL TESTS PASSED, exit 0

## 검증 근거

- **훅 전용 테스트**: `bash tests/hooks/advisory-nudge/test_output_block_falsify_advisory.sh` → **88/88 통과** (rebase 이후 재확인 완료)
- **저장소 전체 테스트**: `bash scripts/run-tests.sh` → **ALL TESTS PASSED, EXIT:0** (라운드 12 수정 이후 최종 재실행)
- 각 Codex 발견은 적용 전 Python 직접 호출로 라이브 프로브 재현 → 수정 → 재프로브로 재확인 → 해당 시나리오의 회귀 테스트를 훅 스위트에 추가하는 순서로 처리했습니다.

## 미검증/남겨진 항목

- 위 헤드라인대로, **round 13 미완료로 convergence 미확인**입니다.
- Codex 자체 sandbox 실행 로그에 반복적으로 나타난 `bash tests/hooks/.../test_output_block_falsify_advisory.sh` 관련 "Command failed (exit 1)"은 라운드 8~13에 걸쳐 매번 동일하게 나타났으나, 매 라운드 로컬에서 직접 재실행하여 실제로는 exit 0·전체 통과임을 확인했습니다 — Codex sandbox의 임시 파일 쓰기 제약에서 비롯된 것으로 보이며 실제 테스트 실패가 아닙니다.

## Summary by CodeRabbit

- **New Features**
  - Improved output validation for AskUserQuestion responses, covering multiple questions and triggering options.
  - Added ready-to-fill `Falsified:` guidance when required evidence is missing.
  - Strengthened detection against incomplete evidence, placeholder text, label mismatches, and formatting bypasses.
  - Added rich telemetry for block and ask decisions while preventing duplicate records.

- **Documentation**
  - Updated behavioral specifications with validation rules, telemetry details, and expanded examples.

- **Tests**
  - Added extensive coverage for multi-question handling, evidence matching, edge cases, and telemetry accuracy.

### CODE REVIEW

File: hooks/advisory-nudge/output-block-falsify-advisory/impl.py [hunk 1]
```diff
@@ -174,13 +178,171 @@ def _has_exact_recommended_marker(labels: list[str]) -> bool:
...
+
+            if matched_label is not None:
+                marker_at = line.find(_PROBE_MARKER, label_end)
+            else:
+                marker_at = line.rfind(_PROBE_MARKER)
+
+            if marker_at != -1:
+                evidence_region = line[marker_at + len(_PROBE_MARKER) :]
+                if any(token in evidence_region for token in _SCAFFOLD_PLACEHOLDER_TOKENS):
+                    return False
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply hooks/advisory-nudge/output-block-falsify-advisory/impl.py:346]: Confirmed via live probe before fixing — an unfilled `Falsified: <label> — probe:` line (nothing, or only whitespace, after the marker) passed the gate.

Fixed in 30bf1da: `evidence_region` is now rejected when `not evidence_region.strip()`, same as an unfilled placeholder. Added 2 regression tests (empty evidence, whitespace-only evidence) — 91/91 hook tests pass, full repo suite (`scripts/run-tests.sh`) passes.

File: hooks/advisory-nudge/output-block-falsify-advisory/impl.py [hunk 2]
```diff
@@ -207,6 +369,114 @@ def _has_recommended_marker(labels: list[str]) -> bool:
...
+    COARSE "pass" record for the same call, turning one deny into
+    `fires=2, block=1, pass=1` and corrupting the exact block-rate count
+    this telemetry exists to provide.
+    """
+    if not isinstance(session_id, str) or not session_id:
+        return
+    _fire_ledger.record_session_fire(
+        _TELEMETRY_HOOK_NAME, _TELEMETRY_HOOK_ROLE, decision, session_id, "AskUserQuestion"
+    )
+    _fire_ledger.suppress_coarse_duplicate()
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply hooks/advisory-nudge/output-block-falsify-advisory/impl.py]: Verified the premise before applying: `record_session_fire` already coerces a non-str `session_id` to `""`, and `aggregate_fires()` only adds a session to its per-session set when `isinstance(session, str) and session` (confirmed by reading both functions directly) — so emitting the RICH record with `session_id=""` is safe and strictly more correct than skipping it, since the original early-return dropped the decision from `fires`/`block`/`ask` totals entirely rather than merely mislabeling it.

Fixed in 30bf1da along the lines of the suggested diff: `_record_block_telemetry` now writes the RICH record unconditionally, with coarse-duplicate suppression as its first, unconditional statement. Did not add a separate gate/reason field — the existing `decision` param (block/ask) already distinguishes T1/T2, and `aggregate_fires` has no consumer for an additional field, so that part of the suggestion was scoped out as unverified/YAGNI rather than applied speculatively.

Updated the missing-session_id test to assert a RICH record with `session_id=""` and `decision=block`, plus a coarse-suppression check. 91/91 hook tests pass, full repo suite passes.
